core/commands/pin/remotepin.go

```diff
@@ -0,0 +1,558 @@
+`,
+		LongDescription: `
+Returns a list of objects that are pinned to a remote pinning service.
+`,
+	},
+
+	Arguments: []cmds.Argument{},
+	Options: []cmds.Option{
+		cmds.StringOption(pinNameOptionName, "Return pins objects with names that contain provided value (case-sensitive, exact match)."),
+		cmds.StringsOption(pinCIDsOptionName, "Return only pin objects for the specified CID(s); optional, comma separated."),
+		cmds.StringsOption(pinStatusOptionName, "Return only pin objects with the specified statuses; optional, comma separated."),
+		cmds.StringOption(pinServiceNameOptionName, "Name of the remote pinning service to use."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx, cancel := context.WithCancel(req.Context)
+		defer cancel()
+
+		service, _ := req.Options[pinServiceNameOptionName].(string)
+		c, err := getRemotePinServiceOrEnv(env, service)
+		if err != nil {
+			return err
+		}
+
+		psCh, errCh, err := lsRemote(ctx, req, c)
+		if err != nil {
+			return err
+		}
+
+		for ps := range psCh {
+			if err := res.Emit(&RemotePinOutput{
+				RequestID: ps.GetRequestId(),
+				Name:      ps.GetPin().GetName(),
+				Delegates: multiaddrsToStrings(ps.GetDelegates()),
+				Status:    ps.GetStatus().String(),
+				Cid:       ps.GetPin().GetCid().String(),
+			}); err != nil {
+				return err
+			}
+		}
+
+		return <-errCh
+	},
+	Type: RemotePinOutput{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *RemotePinOutput) error {
+			fmt.Printf("RequestID: %v\n", out.RequestID)
+			fmt.Printf("CID: %v\n", out.Cid)
+			fmt.Printf("Name: %v\n", out.Name)
+			fmt.Printf("Status: %v\n", out.Status)
+			fmt.Printf("Delegates: %v\n", strings.Join(out.Delegates, ", "))
+			return nil
+		}),
+	},
+}
+
+func lsRemote(ctx context.Context, req *cmds.Request, c *pinclient.Client) (chan pinclient.PinStatusGetter, chan error, error) {
+	opts := []pinclient.LsOption{}
+	if name, nameFound := req.Options[pinNameOptionName].(string); nameFound {
+		opts = append(opts, pinclient.PinOpts.FilterName(name))
+	}
+
+	// TODO: remove debug after ergonomics below are fixed
+	// data, _ := json.Marshal(req.Options)
+	// fmt.Println(string(data))
+
+	if cidsRaw, cidsFound := req.Options[pinCIDsOptionName].([]string); cidsFound {
+		parsedCIDs := []cid.Cid{}
+		// TODO: improve ergonomics: support both --cid=a --cid=b and --cid=a,b
+		for _, rawCID := range cidsRaw {
+			parsedCID, err := cid.Decode(rawCID)
+			if err != nil {
+				return nil, nil, fmt.Errorf("CID %s cannot be parsed (%v)", rawCID, err)
+			}
+			parsedCIDs = append(parsedCIDs, parsedCID)
+		}
+		opts = append(opts, pinclient.PinOpts.FilterCIDs(parsedCIDs...))
+	}
+	if statusRaw, statusFound := req.Options[pinStatusOptionName].([]string); statusFound {
+		parsedStatuses := []pinclient.Status{}
+		// TODO: improve ergonomics: support both --status=a --status=b and --status=a,b
```

Comment:
I switched sharness to use that because `--status=a,b` is not supported by `StringsOption`, it produced 
`statusRaw` with single element `["a,b"]` instead of `["a", "b"]`

Reply:
fixed in f75a209